textpattern/setup/txpsql.php

```diff
@@ -51,130 +51,130 @@
-    $create_sql[] = "INSERT INTO `".PFX."txp_form`(type, name, Form) VALUES('".doSlash($type)."', '".doSlash($name)."', '".doSlash($data)."')";
+    safe_query("INSERT INTO `".PFX."txp_form`(type, name, Form) VALUES('".doSlash($type)."', '".doSlash($name)."', '".doSlash($data)."')");
 }
 
 foreach (get_files_content($themedir.'/pages', 'txp') as $key=>$data) {
-    $create_sql[] = "INSERT INTO `".PFX."txp_page`(name, user_html) VALUES('".doSlash($key)."', '".doSlash($data)."')";
+    safe_query("INSERT INTO `".PFX."txp_page`(name, user_html) VALUES('".doSlash($key)."', '".doSlash($data)."')");
 }
 
-include txpath.'/lib/prefs.php';
 
-foreach ($default_prefs as $name => $p) {
-    create_pref($name, $p[4], $p[0], $p[1], $p[3], $p[2]);
-}
 
-create_user($_SESSION['name'], $_SESSION['email'], $_SESSION['pass'], $_SESSION['realname'], 1);
+// FIXME: Make some wrapper for article and move data to $themedir/articles folder
+$textile = new \Netcarver\Textile\Parser();
 
+$article['body']    = file_get_contents(txpath.DS.'setup'.DS.'article.body.textile');
+$article['excerpt'] = file_get_contents(txpath.DS.'setup'.DS.'article.excerpt.textile');
+$article = str_replace('siteurl', $urlpath, $article);
+$article['body_html']    = $textile->textileThis($article['body']);
+$article['excerpt_html'] = $textile->textileThis($article['excerpt']);
+$article = doSlash($article);
+
+safe_query("INSERT INTO `".PFX."textpattern` VALUES (1, NOW(), NULL, '".doSlash($_SESSION['name'])."', NOW(), '', 'Welcome to your site', '', '".$article['body']."', '".$article['body_html']."', '".$article['excerpt']."', '".$article['excerpt_html']."', '', 'hope-for-the-future', 'meaningful-labor', 1, '".$setup_comment_invite."', 1, 4, '1', '1', 'articles', '', '', '', 'welcome-to-your-site', '', '', '', '', '', '', '', '', '', '', '".md5(uniqid(rand(), true))."', NOW())");
+// Article hardcode end
+
+
+// FIXME: Need some check
 $GLOBALS['txp_install_successful'] = true;
 $GLOBALS['txp_err_count'] = 0;
 $GLOBALS['txp_err_html'] = '';
 
-foreach ($create_sql as $query) {
-    $result = safe_query($query);
 
-    if (!$result) {
-        $GLOBALS['txp_err_count']++;
-        $GLOBALS['txp_err_html'] .= '<li>'.n.
-            '<b>'.htmlspecialchars(mysqli_error($DB->link)).'</b><br />'.n.
-            '<pre>'.htmlspecialchars($query).'</pre>'.n.'</li>'.n;
-        $GLOBALS['txp_install_successful'] = false;
-    }
-}
-
-// Reduild categories tree
+// Funal reduild category trees
 rebuild_tree_full('article');
 rebuild_tree_full('link');
 rebuild_tree_full('image');
 rebuild_tree_full('file');
 
 
-require_once txpath.'/lib/IXRClass.php';
-$client = new IXR_Client('http://rpc.textpattern.com');
 
-if (!$client->query('tups.getLanguage', $blog_uid, LANG)) {
-    // If cannot install from lang file, setup the English lang.
-    if (!install_language_from_file(LANG)) {
-        $lang = 'en-gb';
-        include_once txpath.'/setup/en-gb.php';
+function setup_txp_lang($lang)
```

Comment:
Don't worry too much about refactoring any of this RPC stuff. The RPC reliance is (hopefully) disappearing in this release and we'll install lang strings from files bundled in core. See the languages-inclusion branch for more.